code/game/objects/items/weapons/switchtool.dm

```diff
@@ -207,3 +216,178 @@
 	if(istype(deployed, /obj/item/weapon/lighter))
 		var/obj/item/weapon/lighter/lighter = deployed
 		lighter.lit = 1
+		..()
+
+//Unique RD switchtool, modules cannot be removed nor inserted to upgrade, but require techdisks to aquire new modules.
+/obj/item/weapon/switchtool/holo
+	name = "Holo switchtool"
+	icon_state = "holo_switchtool"
+	item_state = "Hswitchtool"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/switchtools.dmi', "right_hand" = 'icons/mob/in-hand/right/switchtools.dmi')
+	desc = "A switchtool that can take on the form of nearly any tool. It's experimental hardlight emitter requires tech disks to help define it's shape."
+	var/brightness_max = 4
+	var/brightness_min = 2
+	deploy_sound = "sound/weapons/switchsound.ogg"
+	undeploy_sound = "sound/weapons/switchsound.ogg"
+	//
+	//deploy_sound = "sound/weapons/saberon.ogg"
+	//undeploy_sound = "sound/weapons/saberoff.ogg"
+	light_color =  LIGHT_COLOR_CYAN
+	no_remove = 1
+	mech_flags = MECH_SCAN_ILLEGAL
+
+	stored_modules = list(//scalpel and flashlight are available to start and the scalpel is logically a laser one but the basic kind.
+						"/obj/item/device/flashlight:Light" = null,
+						"/obj/item/weapon/scalpel/laser/tier1:Scalpel" = null)
+
+//Checks the research type and level for the respective field, then adds them all to the stored modules while also filling the slot with that tool.
+/obj/item/weapon/switchtool/holo/attackby(var/obj/item/weapon/D as obj, var/mob/user as mob)
+	if(..())
+		return
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/biotech) && disk_tech.level >= 3)
+			stored_modules["/obj/item/weapon/circular_saw:Circular saw"] = new /obj/item/weapon/circular_saw(src)
+			stored_modules["/obj/item/weapon/surgicaldrill:Surgical drill"] = new /obj/item/weapon/surgicaldrill(src)
+			stored_modules["/obj/item/weapon/cautery:Cautery"] = new /obj/item/weapon/cautery(src)
+			stored_modules["/obj/item/weapon/hemostat:Hemostat"] = new /obj/item/weapon/hemostat(src)
+			stored_modules["/obj/item/weapon/retractor:Retractor"] = new /obj/item/weapon/retractor(src)
+			stored_modules["/obj/item/weapon/bonesetter:Bonesetter"] = new /obj/item/weapon/bonesetter(src)
+			to_chat(user, "The Holo switchtool has medical designs now!")
+			return
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
```

Comment:
You don't need to do the same `istype()` check over and over again. You can just put all the `disk_tech` checks underneath the first `if(istype(D, /obj/item/weapon/disk/tech_disk))`.